ambry-protocol/src/main/java/com.github.ambry.protocol/MessageInfoListSerde.java

```diff
@@ -78,14 +96,32 @@ void serializeMessageInfoList(ByteBuffer outputBuffer) {
         outputBuffer.putLong(messageInfo.getSize());
         outputBuffer.putLong(messageInfo.getExpirationTimeInMs());
         outputBuffer.put(messageInfo.isDeleted() ? DELETED : (byte) ~DELETED);
-        if (version == MessageInfoListVersion_V2) {
-          Long crc = messageInfo.getCrc();
-          if (crc != null) {
-            outputBuffer.put(CRC_PRESENT);
-            outputBuffer.putLong(crc);
-          } else {
-            outputBuffer.put((byte) ~CRC_PRESENT);
-          }
+        switch (version) {
+          case VERSION_1:
+            break;
+          case VERSION_2:
+            Long crc = messageInfo.getCrc();
```

Comment:
minor could you move the declaration `Long crc;` outside of the switch and just do assignments inside of the switch.